tests/scripts/generate_tests_pk_can_do_psa.py

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import re
+
+# PK contexts to be tested
+PK_CONTEXTS = {
+    'MBEDTLS_PK_ECDSA': {
+        'PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS': {
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH)': [
```

Comment:
We should test both PSA variants, not just the one that PK uses by default. That's important for wrapped PSA keys.

Given that the primary motivation for `mbedtls_pk_can_do_psa` is TLS key checks and [it already has a bug in this respect](https://github.com/Mbed-TLS/mbedtls/issues/10220), we can omit testing the new function in this respect for now, and file a subsequent task to test it before we fix the TLS consumer.